service/src/java/org/apache/hive/service/auth/ldap/CustomQueryFilterFactory.java

```diff
@@ -76,6 +76,19 @@ public void apply(DirSearch client, String user) throws AuthenticationException
           }
         }
       }
+
+      // try a generic user search
+      String userSearchQuery = query.replace("{0}", user);
```

Comment:
if the query does not contain '{0}',  the `userSearchQuery` would be the same as `query`, which makes TestLdapAtnProviderWithMiniDS#testCustomQueryNegative fail in this case.

Reply:
Resolved